pyproject.toml

```
@@ -18,18 +18,16 @@ dependencies = [
     "Quart",
     "aiohttp",
     "pydantic-settings",
-    "python-dotenv",
     "granian",
     "transformers",
     "torch",
     "protobuf",
-    "python-dotenv",
     "sentencepiece"
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
-dev = ["pytest", "ruff", "pytest-asyncio"]
+dev = ["pytest", "ruff", "pytest-asyncio", "python-dotenv"]
```

```suggestion
dev = ["pytest", "ruff", "pytest-asyncio"]
```
python-dotenv can actually be dropped completly, it is only used as part of pydantic-settings which includes it as a dep anyway: https://github.com/pydantic/pydantic-settings/blob/main/pyproject.toml#L44

---

@b3n4kh Maybe a silly question, but why would this be a good practice? Reliance on `pydantic` sounds unnatural to me, especially because we introduce it here directly to load variables for the bot.

---

If it is used without pydantic it could make sense to list it specifically, but I don't know where we are using it outside of pydantic, can you point me to the code?

---

It is solely used in conftest.py, to load a defined .env.test (https://github.com/taranis-ai/summarize_bot/blob/fix/use_max_new_tokens/summarize_bot/tests/conftest.py)
This is a security measure, so you don't actually use a local .env file and wonder why tests are failing (like i did)
So it should be a dev dependency

---

I see yes then it should add it to dev dependencies!

Although in the future we could look into overriding [SettingsConfigDict](https://github.com/taranis-ai/summarize_bot/blob/fix/use_max_new_tokens/summarize_bot/config.py#L8) rather then using load_dotenv, which would probably be cleaner.

---

Ah, I see, correct - a dev dependency would be best for this.

---

thank you